### PR TITLE
executor: do not use batchChecker in 'insert ignore into ...'

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

Continue with https://github.com/pingcap/tidb/pull/12108, please review this PR after it's merged.

After this commit, the insert operations do not use the `batchChecker` anymore.

### What is changed and how it works?

Refactoring code.
Implement the 'insert ignore into ...' operation without the `batchChecker`

### Check List 

Tests 

 - No code

Release note

 - Write release note for bug-fix or new feature.

### CODE REVIEW

**Comment by jackysp** [comment]
Please resolve the conflicts.

File: executor/insert.go [hunk 1]
```diff
@@ -149,9 +149,9 @@ func prefetchDataCache(ctx context.Context, txn kv.Transaction, rows []toBeCheck
 	return prefetchConflictedOldRows(ctx, txn, rows, values)
 }
 
-// updateDupRowNew updates a duplicate row to a new row.
-func (e *InsertExec) updateDupRowNew(ctx context.Context, txn kv.Transaction, row toBeCheckedRow, handle int64, onDuplicate []*expression.Assignment) error {
-	oldRow, err := e.getOldRowNew(ctx, e.ctx, txn, row.t, handle, e.GenExprs)
```
Comment (lysu) [inline executor/insert.go]: so `batchChecker) getOldRowNew` can be remove now~?

Comment (tiancaiamao) [reply executor/insert.go]: Not now, the `replace ...` is still using it.
I'll remove it later...

Comment (lysu) [reply executor/insert.go]: https://github.com/pingcap/tidb/blob/dd11b55699c615377a481667eef76b5fa5ae03a7/executor/batch_checker.go#L297 this method can not find any usage :D

**Pull request merged**